src/mlpack/methods/hmm/hmm_train_main.cpp

```diff
@@ -379,6 +379,12 @@ static void mlpackMain()
         "unknown HMM type");
   }
 
+  if (CLI::HasParam("tolerance"))
+  {
+    RequireParamValue<int>("tolerance", [](int x) { return x >= 0; }, true,
+        "tolerance must be non-negative");
+  }
```

Comment:
Thanks for adding this.  I think the `if (CLI::HasParam("tolerance"))` is unnecessary though---if the parameter isn't passed, it will be the default, which is greater than 0.

Reply:
Yes, you're right. Also I think that the template param here should be double instead of int -- will fix it.